setup.py

```diff
@@ -27,6 +28,32 @@ def load_module_from_path(module_name, path):
 ROOT_DIR = os.path.dirname(__file__)
 logger = logging.getLogger(__name__)
 
+
+def embed_commit_hash():
+    try:
+        commit_id = subprocess.check_output(
+            ['git', 'rev-parse', '--short', 'HEAD'], encoding="utf-8").strip()
+
+        version_file = os.path.join(ROOT_DIR, "vllm", "version.py")
+        with open(version_file, encoding="utf-8") as f:
+            version_contents = f.read()
+
+        version_contents = version_contents.replace("None", f'"{commit_id}"')
```

Comment:
it might be better have a more obnoxious replacement target ... maybe something like "UNSPECIFIED-HASH"

Reply:
I suppose since this is now part of `setup.py`, that is more reasonable, as even a local install will inject the hash (unless basically `git` is unavailable). At a bare minimum, I like the placeholder being a string value for type consistency. I’ve updated it to `"COMMIT_HASH_PLACEHOLDER"`.